src/track/cueinfoimporter.cpp

```diff
@@ -0,0 +1,79 @@
+#include "track/cueinfoimporter.h"
+
+namespace mixxx {
+
+CueInfoImporter::CueInfoImporter() {
+}
+
+CueInfoImporter::CueInfoImporter(const QList<CueInfo>& cueInfos)
+        : m_cueInfos(cueInfos) {
+}
+
+double CueInfoImporter::getTimingOffsetMillis(
+        const QString& filePath,
+        const audio::SignalInfo& signalInfo) const {
+    Q_UNUSED(filePath);
+    Q_UNUSED(signalInfo);
+    return 0;
+};
+
+void CueInfoImporter::append(const CueInfo& cueInfo) {
+    m_cueInfos.append(cueInfo);
+}
+
+void CueInfoImporter::append(const QList<CueInfo>& cueInfos) {
+    m_cueInfos.append(cueInfos);
+}
+
+int CueInfoImporter::size() const {
+    return m_cueInfos.size();
+}
+
+bool CueInfoImporter::isEmpty() const {
+    return m_cueInfos.isEmpty();
+}
+
+QList<CueInfo> CueInfoImporter::getCueInfosWithCorrectTiming(
+        const QString& filePath,
+        const audio::SignalInfo& signalInfo) {
+    QList<CueInfo> cueInfos;
```

Comment:
We can pick the internal list and don't need to create a new list.

Try this:
```
QList<CueInfo> cueInfos(std::move(m_cueInfos));
DEBUG_ASSERT(m_cueInfos.isEmpty());
```

The debug assertion simply documents our intention.

Reply:
Is this okay, too? https://github.com/mixxxdj/mixxx/pull/2707/files#diff-a90e67d47a0490f3d58383410d1b5a2eR48

Reply:
We still don't need this separate list and can import m_cueInfos directly.

Reply:
I don't understand, what separate list?

```cpp
// This does involve a separate list
QList<CueInfo> cueInfos(std::move(m_cueInfos));
DEBUG_ASSERT(m_cueInfos.isEmpty());
return cueInfos;

// This doesn't, right?
return std::move(m_cueInfos);
```

Reply:
https://github.com/uklotzde/mixxx/commit/fd1a544e670eae7d9a4af66c974e39020e8a258c

Reply:
Sorry, forgot to remove newCueInfo

Reply:
https://github.com/uklotzde/mixxx/commit/749790baeba370b583a0af254d4ddb5efd96c28b